### ISSUE TITLE
UI - Subject area UI fails with NPE

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#1759]
Regression (has worked in the last week or so)
* docker-compose environment (15 Oct docker images/master)
* Environment:
            - OMAS_SERVER_URL=http://datalake:8080
            - OMAS_SERVER_NAME=cocoMDS4
* Login to UI using faithbroker/admin at https://localhost:18443 (as setup by docker-compose)
* Navigate to subject area, hit + against project and use 'aaa' 'bbb' 'ccc' as values & click create
* Popup 'we are experiencing an unexpected error' Rotating progress indicator starts, never completes
* No exceptions noticed on server platforms
* UI reports exception as:
```
2019-10-15 11:55:15.772  INFO 1 --- [nio-8443-exec-2] o.s.web.servlet.DispatcherServlet        : Completed initialization in 80 ms
2019-10-15 11:55:18.086 ERROR 1 --- [nio-8443-exec-5] o.o.o.u.accessservices.auth.AuthFilter   : TOKEN EXPIRED
2019-10-15 11:55:18.213 ERROR 1 --- [nio-8443-exec-5] o.o.o.u.accessservices.auth.AuthFilter   : TOKEN EXPIRED
2019-10-15 11:55:22.592  INFO 1 --- [nio-8443-exec-7] o.h.h.i.QueryTranslatorFactoryInitiator  : HHH000397: Using ASTQueryTranslatorFactory
2019-10-15 11:55:26.158 ERROR 1 --- [nio-8443-exec-3] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.NullPointerException] with root cause

java.lang.NullPointerException: null
    at org.odpi.openmetadata.userinterface.accessservices.api.SecureController.getUser(SecureController.java:24) ~[classes!/:na]
    at org.odpi.openmetadata.userinterface.accessservices.api.SubjectAreaProjectController.findProject(SubjectAreaProjectController.java:146) ~[classes!/:na]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_212]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_212]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_212]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_212]
    at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190) ~[spring-web-5.1.8.RELEASE.jar!/:5.1.8.RELEASE]
```
Full log at : https://gist.github.com/1add9bce2c60c7bc2ec71b8c244abfea

This may have been introduced by #1698 cc: @ruxandraRosu @bogdan-sava @davidradl @moyabrannan

**Comment by ruxandraRosu** [comment]
Hi David, this is probably caused by my changes, sorry for it. I've added a property to allow for authentication either by token or session. This means that user is no longer on session if application is configured to run using tokens. I've done a similar change in UserInfoController, sorry for missing your controller.
Do you want me to fix this or are you working on it?

**Comment by davidradl** [comment]
@ruxandraRosu I am happy for you to fix it - thanks :-).  I was thinking it would be worth adding protection code to SecurityController so it only addresses the user if it is not null - to prevent the NPE. Does the type explorer need to be changed as well? 

Out of interest, if it configured to use token, where does it now get the user from?

**Comment by ruxandraRosu** [comment]
Hi, I have a fix for this on dev-ui branch, I'll create a PR for it. I don't think type explorer is taking the user from session, so no changes needed there. For token based, user details are in token.

**Comment by davidradl** [comment]
@ruxandraRosu Ruxandra, Great :-) are you ok to put in a pr for it and I will review it, so I can see what you changed,   all the best, David.

**Comment by planetf1** [comment]
Thanks - I am hoping to make the release 1.1 branch thu or fri, I think this fix should be in 1.1 if at all possible. I will also test in the compose environment once merged

**Comment by planetf1** [comment]
Proposed fix for this is #1766

**Comment by planetf1** [comment]
Closing as now fixed
